src/autodiff/cppad.hpp

```diff
@@ -28,7 +28,14 @@ namespace boost
       namespace detail
       {
         template<typename Scalar>
-        struct constant_pi< CppAD::AD<Scalar> > : constant_pi<Scalar> {};
+        struct constant_pi< CppAD::AD<Scalar> > : constant_pi<Scalar> {
+          template <int N>
+          static inline CppAD::AD<Scalar> get(const mpl::int_<N>& n)
```

Comment:
Small question: why do you need this?

Reply:
When explog is called with the CppAD class, it calls for PI. This introduces an error.